cmd/collectors/zapi/collector/zapi.go

```diff
@@ -269,28 +271,28 @@ func (me *Zapi) PollData() (*matrix.Matrix, error) {
 	fetch = func(instance *matrix.Instance, node *node.Node, path []string) {
 
 		newpath := append(path, node.GetNameS())
-		logger.Debug(me.Prefix, " > %s(%s)%s <%s%d%s> name=[%s%s%s%s] value=[%s%s%s]", color.Grey, newpath, color.End, color.Red, len(node.GetChildren()), color.End, color.Bold, color.Cyan, node.GetNameS(), color.End, color.Yellow, node.GetContentS(), color.End)
+		key := strings.Join(newpath, ".")
+		logger.Debug(me.Prefix, " > %s(%s)%s <%s%d%s> name=[%s%s%s%s] value=[%s%s%s]", color.Grey, key, color.End, color.Red, len(node.GetChildren()), color.End, color.Bold, color.Cyan, node.GetNameS(), color.End, color.Yellow, node.GetContentS(), color.End)
 
 		if value := node.GetContentS(); value != "" {
-			key := strings.Join(newpath, ".")
-			if metric := me.Matrix.GetMetric(key); metric != nil {
+			if label, has := me.instanceLabelPaths[key]; has {
+				instance.SetLabel(label, value)
+				logger.Trace(me.Prefix, "%slabel (%s) [%s] set value (%s)%s", color.Yellow, key, label, value, color.End)
```

Comment:
did you mean to leave this uncommented

Reply:
no, I uncommented because it might help to understand other issues in the future